src/compaction_filter.rs

```diff
@@ -70,7 +94,18 @@ pub unsafe fn new_compaction_filter(
     c_name: CString,
     ignore_snapshots: bool,
     f: Box<dyn CompactionFilter>,
-) -> Result<CompactionFilterHandle, String> {
+) -> CompactionFilterHandle {
+    let filter = new_compaction_filter_raw(c_name, ignore_snapshots, f);
+    CompactionFilterHandle { inner: filter }
+}
+
+/// Just like `new_compaction_filter`, but returns a raw pointer instead of a RAII struct.
+/// Generally used in `CompactionFilterFactory::create_compaction_filter`.
+pub unsafe fn new_compaction_filter_raw(
+    c_name: CString,
+    ignore_snapshots: bool,
```

Comment:
Do me a favor removing `ignore_snapshots`? It's obsolete recently: https://github.com/facebook/rocksdb/blob/master/include/rocksdb/compaction_filter.h#L187

Reply:
It's removed in the lastest commit. Could you help me to check it again? thanks!